src/python_bindings/py_data_stats.cpp

```diff
@@ -0,0 +1,166 @@
+#include "py_data_stats.h"
+
+#include "statistics/data_stats.h"
+
+namespace python_bindings {
+
+namespace py = pybind11;
+namespace mo = model;
+
+size_t PyDataStats::GetNumberOfValues(size_t index) const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).NumberOfValues(index);
+}
+
+size_t PyDataStats::GetNumberOfColumns() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetNumberOfColumns();
+}
+
+std::vector<size_t> PyDataStats::GetNullColumns() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetNullColumns();
+}
+
+std::vector<size_t> PyDataStats::GetColumnsWithNull() const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).GetColumnsWithNull();
+}
+
+std::vector<size_t> PyDataStats::GetColumnsWithUniqueValues() const {
+    return dynamic_cast<algos::DataStats&>(*algorithm_).GetColumnsWithUniqueValues();
```

Comment:
All `dynamic_cast`s should be replaced with `GetAlgorithm`

Reply:
Actually, all these casts look a bit unreasonable since we know the specific algorithm type in this context. Seems like `algorithm_` should be moved to `PyAlgorithm`, however, this is for a different PR

Reply:
Since these stat methods can't be called via a const object I can't use `GetAlgoritm` there

Reply:
Changing `T const&` to `T&` in `GetAlgorithm` should work.